packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.classNames.ts

```diff
@@ -0,0 +1,97 @@
+import { memoizeFunction } from '../../Utilities';
+import { mergeStyles, IStyle } from '../../Styling';
+import { ISpinButtonStyles } from './SpinButton.Props';
+import { KeyboardSpinDirection } from './SpinButton';
+import { Position } from '../../utilities/positioning';
+
+export interface ISpinButtonClassNames {
+  root: string;
+  labelWrapper: string;
+  icon: string;
+  label: string;
+  spinButtonWrapper: string;
+  input: string;
+  arrowBox: string;
+}
+
+export const getClassNames = memoizeFunction((
+  styles: ISpinButtonStyles,
+  disabled: boolean,
+  isFocused: boolean,
+  keyboardSpinDirection: KeyboardSpinDirection,
+  labelPosition: Position = Position.start
+): ISpinButtonClassNames => {
+  return {
+    root: mergeStyles(
+      styles.root
+    ) as string,
+    labelWrapper: mergeStyles(
+      styles.labelWrapper,
+      _getStyleForLabelBasedOnPosition(labelPosition, styles)
+    ) as string,
+    icon: mergeStyles(
+      styles.icon,
+    ) as string,
+    label: mergeStyles(
+      styles.label
+    ) as string,
+    spinButtonWrapper: mergeStyles(
+      styles.spinButtonWrapper,
+      _getStyleForRootBasedOnPosition(labelPosition, styles),
+      !disabled && [
+        {
+          ':hover': styles.spinButtonWrapperHovered
+        },
+        isFocused && {
+          // This is to increase the specifity of the focus styles
+          // and make it equal to that of the hover styles.

```

Comment:
We have the classNames file to ensure the order - it does put the focus styles after the hover styles here.. The problem was regardless of the order, the specificity of the hover styles was always greater than that of the focused styles. Therefore, I have had to resort to this.